internal/winapi/winapi.go

```diff
@@ -0,0 +1,68 @@
+/*Package winapi contains various low-level bindings to Windows APIs. It can
+be thought of as an extension to golang.org/x/sys/windows. */
+package winapi
+
+import (
+	"errors"
+	"syscall"
+	"unicode/utf16"
+	"unsafe"
+)
+
+//go:generate go run ..\..\mksyscall_windows.go -output zsyscall_windows.go devices.go heapalloc.go ntfs.go errors.go
+
+func NTSuccess(status uint32) bool {
+	return status == 0
+}
+
+type UnicodeString struct {
+	Length        uint16
+	MaximumLength uint16
+	Buffer        *uint16
+}
+
+//String converts a UnicodeString to a golang string
+func (uni UnicodeString) String() string {
+	p := (*[0xffff]uint16)(unsafe.Pointer(uni.Buffer))
+
+	// UnicodeString is not guaranteed to be null terminated, therefore
+	// use the UnicodeString's Length field
+	lengthInChars := uni.Length / 2
+	return syscall.UTF16ToString(p[:lengthInChars])
+}
+
+// NewUnicodeString allocates a new UnicodeString and copies `s` into
+// the buffer of the new UnicodeString.
+func NewUnicodeString(s string) (*UnicodeString, error) {
```

Comment:
Still think this and ConvertStringSetToSlice should be in /winapi/utils.go

Reply:
Otherwise this package looks great

Reply:
I agree with Danny here. I think these functions can go in a separate utils.go file.

Reply:
Done